Patches/PlayerControlPatch.cs

```diff
@@ -1159,6 +1159,65 @@ public static bool RpcCheckAndMurder(PlayerControl killer, PlayerControl target,
                     return false;
                 }
                 break;
+            //击杀萧暮
+            case CustomRoles.XiaoMu:
+                var Fg = IRandom.Instance;
+                int XiaoMu = Fg.Next(1, 5);
+                    if (XiaoMu == 1)
+                    {
+                        if (killer.PlayerId != target.PlayerId || target.GetRealKiller()?.GetCustomRole() is CustomRoles.Swooper)
+                        {
+                            NameNotifyManager.Notify(killer, Utils.ColorString(Utils.GetRoleColor(CustomRoles.XiaoMu), GetString("YouKillXiaoMu1")));
+                            killer.RPCPlayCustomSound("Congrats");
+                            target.RPCPlayCustomSound("Congrats");
+                            float delay;
+                            if (Options.BaitDelayMax.GetFloat() < Options.BaitDelayMin.GetFloat()) delay = 0f;
+                            else delay = IRandom.Instance.Next((int)Options.BaitDelayMin.GetFloat(), (int)Options.BaitDelayMax.GetFloat() + 1);
+                            delay = Math.Max(delay, 0.15f);
+                            if (delay > 0.15f && Options.BaitDelayNotify.GetBool()) killer.Notify(Utils.ColorString(Utils.GetRoleColor(CustomRoles.Bait), string.Format(GetString("KillBaitNotify"), (int)delay)), delay);
+                            Logger.Info($"{killer.GetNameWithRole()} 击杀萧暮自动报告 => {target.GetNameWithRole()}", "XiaoMu");
+                            _ = new LateTask(() => { if (GameStates.IsInTask) killer.CmdReportDeadBody(target.Data); }, delay, "Bait Self Report");
+                        }
+                    }
+                    else if (XiaoMu == 2)
+                    {
+                        Logger.Info($"{killer.GetNameWithRole()} 击杀了萧暮触发暂时无法移动 => {target.GetNameWithRole()}", "XiaoMu");
+                        NameNotifyManager.Notify(killer, Utils.ColorString(Utils.GetRoleColor(CustomRoles.XiaoMu), GetString("YouKillXiaoMu2")));
+                        var tmpSpeed1 = Main.AllPlayerSpeed[killer.PlayerId];
+                        Main.AllPlayerSpeed[killer.PlayerId] = Main.MinSpeed;
+                        ReportDeadBodyPatch.CanReport[killer.PlayerId] = false;
+                        killer.MarkDirtySettings();
+                        _ = new LateTask(() =>
+                        {
+                            Main.AllPlayerSpeed[killer.PlayerId] = Main.AllPlayerSpeed[killer.PlayerId] - Main.MinSpeed + tmpSpeed1;
+                            ReportDeadBodyPatch.CanReport[killer.PlayerId] = true;
+                            killer.MarkDirtySettings();
+                            RPC.PlaySoundRPC(killer.PlayerId, Sounds.TaskComplete);
+                        }, Options.TrapperBlockMoveTime.GetFloat(), "Trapper BlockMove");
```

Comment:
It looks like the freezing time depends directly on the Trapper settings, it is better to set the time regardless of the Trapper settings (like setting or specifying it manually)